### PR TITLE
HITL - Synchronize agent sensors with cameras.

### PR DESCRIPTION
**Pull request by 0mdc**
## Motivation and Context

This changeset synchronizes HITL agent sensors with the HITL-driven camera.

* Add camera-sensor synchronization.
* Add `rearrange_v2` configuration for app-specific settings.
* Add API to find agent name or index.
* Remove hard-coded flag to specify whether an agent can change object states.

## How Has This Been Tested

Tested on single-learn by @zephirefaith 

## Types of changes

- **\[Development\]**

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated the documentation if required.
- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes if required.

### CODE REVIEW

File: examples/hitl/rearrange_v2/rearrange_v2.py [hunk 1]
```diff
@@ -157,30 +163,67 @@ def __init__(
...
+        self, global_cam_transform: mn.Matrix4
+    ) -> None:
+        """
+        Updates the camera transform of the agent.
+        If the agent has 'head sensors', this will also update their transform.
+        """
+        self.cam_transform = global_cam_transform
+
+        for render_camera in self.render_cameras:
+            # TODO: There is currently no utility to set a global transform.
```
Comment (0mdc) [inline examples/hitl/rearrange_v2/rearrange_v2.py:203]: I'm planning to include this to `sim_utilities` or `habitat_sim`.

File: examples/hitl/rearrange_v2/rearrange_v2.py [hunk 2]
```diff
@@ -459,6 +503,38 @@ def _on_error_reported(self, error_report: ErrorReport):
...
+class RearrangeV2AgentConfig:
+    head_sensor_substring: str
+    """Substring used to identify the agent's head sensor."""
+
+    can_change_object_states: bool
+    """Whether the agent can modify object states."""
+
+
+@dataclass
+class RearrangeV2Config:
```
Comment (0mdc) [inline examples/hitl/rearrange_v2/rearrange_v2.py:516]: These will eventually be changed to structured configs. Not bothering to do this here.

**Review by 0mdc — COMMENTED** [review]
Comments for reviewers.

**Review by zephirefaith — APPROVED** [review]
LGTM. Confirming this fixes things for single-learn and WG includes obs from both human and robot.

**Pull request merged**